pyproject.toml

```diff
@@ -27,40 +34,146 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Physics",
     "Topic :: Software Development :: Libraries :: Python Modules"
 ]
+dependencies = [
+    "pydantic~=2.5.2",
+    "h5py",
+    "numpy~=1.26.0",
+    "Pillow~=10.3.0",
```

Comment:
**issue:** version constraint used to be stricter with poetry, using "= ~"

Likely means for Hatch:
```suggestion
    "numpy==1.26.*",
    "Pillow==10.3.*",
```